crates/archive/src/tarball.rs

```diff
@@ -30,6 +30,40 @@ pub struct Tarball {
     origin: Origin,
 }
 
+/// Thrown when the containing directory could not be determined
+#[derive(Fail, Debug)]
+#[fail(display = "Could not determine directory information for {}", path)]
+struct ContainingDirError {
+    path: String,
+}
+
+/// Thrown when the containing directory could not be determined
+#[derive(Fail, Debug)]
+#[fail(display = "Could not create directory {}", dir)]
+struct CreateDirError {
+    dir: String,
+}
+
+/// This creates the parent directory of the input path, assuming the input path is a file.
+pub fn ensure_containing_dir_exists<P: AsRef<Path>>(path: &P) -> Result<(), failure::Error> {
```

Comment:
Is it possible to make this call in `volta-core`, maybe immediately before calling unpack? As opposed to duplicating this function in a different crate?

Reply:
I was actually planning to extract this to a new `fs-utils` crate, so it could be used by any of the crates.

Reply:
That works too 👍 That can be a follow-on issue as well if that’s easier.

Reply:
Went ahead and did it :D